File: FizzBuzz Fun/PROLOG/FizzBuzz.pl
fizzbuzz :-
        foreach(between(1, 100, X), print_item(X)).
 
print_item(X) :-
        (  0 is X mod 15
        -> print('FizzBuzz')
        ;  0 is X mod 3
        -> print('Fizz')
        ;  0 is X mod 5
        -> print('Buzz')
        ;  print(X)
        ),
        nl.
/*for each element 1-100 print based on following criteria,
print FizzBuzz if mod 3 and mod 5
print Fizz if mod 3
print Buzz if mod 5
print current integer if mod neither
*/
    

File: FizzBuzz Fun/HASKELL/FizzBuzz.hs
fizzbuzz n = case (rem n 3, rem n 5) of -- pattern matching approach
	       (0, 0) -> "FizzBuzz" -- both mod 3 and mod 5
               (0, _) -> "Fizz" -- only mod 3
               (_, 0) -> "Buzz" -- only mod 5
               (_, _) -> show n -- not mod 3 or 5, print current integer
 
main = mapM_ (putStrLn . fizzbuzz) [1..100] -- run main

--code based on example from RosettaCode
